{"version":3,"file":"argument-null-exception.js","sourceRoot":"","sources":["../src/argument-null-exception.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,yCAAoC;AACpC,iBAAe;AAEf;IAAoC,yCAAS;IAIzC,+BAAmB,OAAe,EAAE,cAA0B;QAA9D,iBAQC;QANG,EAAE,CAAC,CAAC,OAAO,IAAI,IAAI,IAAI,OAAO,CAAC,mBAAmB,EAAE,CAAC;YACjD,OAAO,GAAG,WAAW,CAAC;QAE1B,IAAM,OAAO,GAAG,yBAAyB,CAAC,MAAM,CAAC,OAAO,CAAC,CAAC;QAE1D,QAAA,kBAAM,OAAO,EAAE,cAAc,CAAC,SAAC;;IACnC,CAAC;IACL,4BAAC;AAAD,CAAC,AAbD,CAAoC,mBAAS,GAa5C;AAED,kBAAe,qBAAqB,CAAC","sourcesContent":["import Exception from \"./exception\";\nimport \"n-ext\";\n\nclass ArgumentNullException extends Exception\n{\n    public constructor(argName: string);\n    public constructor(argName: string, innerException: Exception);\n    public constructor(argName: string, innerException?: Exception)\n    {\n        if (argName == null || argName.isEmptyOrWhiteSpace())\n            argName = \"<UNKNOWN>\";\n\n        const message = \"Argument '{0}' is NULL.\".format(argName);\n\n        super(message, innerException);\n    }\n}\n\nexport default ArgumentNullException;"]}